static/js/compose.js

```diff
@@ -623,7 +584,6 @@ exports.respond_to_message = function (opts) {
     var msg_type;
     // Before initiating a reply to a message, if there's an
     // in-progress composition, snapshot it.
-    compose.snapshot_message();
```

Comment:
If you click on another message, the compose box never loses focus, it just changes state.  That's what this code path is for.